### DISCUSSION TITLE
Site not running anymore after upgrade to Nuxt3 RC3

### DISCUSSION THREAD

**Discussion by nachtfunke** [nuxt/nuxt#17514] [Questions] [answered]
So I worked on a site that had an older version of nuxt3 installed: `3.0.0-27242401.b3ff8ee`. After upgrading to `3.0.0-rc.3`, the site is entirely broken when I start it with `nuxt dev`. I am getting a client-side javascript error:

```javascript
Error while mounting app: TypeError: Cannot read properties of undefined (reading 'modules')
    at _sfc_main.setup (root.vue:17:16)
    at callWithErrorHandling (runtime-core.esm-bundler.js:155:22)
    at setupStatefulComponent (runtime-core.esm-bundler.js:7109:29)
    at setupComponent (runtime-core.esm-bundler.js:7064:11)
    at mountComponent (runtime-core.esm-bundler.js:4951:13)
    at hydrateNode (runtime-core.esm-bundler.js:4153:21)
    at hydrate2 (runtime-core.esm-bundler.js:4068:9)
    at mount (runtime-core.esm-bundler.js:3900:25)
    at Object.app.mount (runtime-dom.esm-bundler.js:1612:20)
    at initApp (entry.mjs:34:12)
```

I have no idea what this error means or how to fix it. I am not sure how to proceed here. I am out of options to fix this myself.

**Comment by danielroe** [comment]
Would you provide some code or a reproduction? 🙏

**Reply by nachtfunke** [reply]
The error does not point me to any confined part of the codebase. I cannot give you a specific piece of code. What you can do reproduce it, is fork the repo and try it out yourself: https://github.com/nachtfunke/psb-sonst-nichts.

I know its difficult to answer these issues in isolation, but the problem presents itself as an isolated problem devoid of context to me as well.

**Reply by danielroe** [reply]
With the repository cloned and the steps below from @viii47 your repo worked fine for me:

<img width="1656" alt="CleanShot 2022-05-27 at 17 06 59@2x" src="https://user-images.githubusercontent.com/28706372/170727114-ffeacc83-1e0d-4e0f-a7b4-6950f047b9ad.png">

The process of switching from edge -> RC was a bit involved as the key dependency name was changed from nuxt3 -> nuxt and therefore required a full purge of lockfile/node_modules/.nuxt etc.

**Comment by viii47** [comment, marked as answer]
- delete `node_modules`, `yarn.lock`, `.nuxt`, `.output`
- edit the import statement inside the **nuxt.config** to `import { defineNuxtConfig } from 'nuxt'`, not `import { defineNuxtConfig } from 'nuxt3'`
- edit the package.json dependency 
- run `yarn cleanup` & `yarn`

If nothing keeps working have a look at the closed issues https://github.com/nuxt/framework/issues?q=is%3Aissue+is%3Aclosed+module+rc.3 - there are multiple issues that can happen on upgrading from a non-rc to rc.3x (import missing / broken, isEqual error, etc.)

**Reply by nachtfunke** [reply]
Thank you so much, this worked. The site is entirely broken now, but at least it runs again. I feel like I can fix everything else now. Thank you so much @viii47

**Reply by lacorde** [reply]
Thank you @viii47, deleting `.nuxt` & `.output` is what saved me.

This part of the process is so easily forgotten.
